### PR TITLE
chore: Move appbar logic to utils.

### LINKED ISSUE #1485: Move action bar logics to utils
**Describe the feature you'd like**
Move action bar logic to utils like - Set title, handle visibility and display back button. No need to cast again and again for each fragment.

**Would you like to work on the issue?**

open for all

### PR DESCRIPTION
**Pull request by angmas1**
Fixes #1485 

Changes: Moves appbar logics to utils.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 1]
```diff
@@ -77,6 +79,14 @@ object Utils {
         inputManager.hideSoftInputFromWindow(view.windowToken, InputMethodManager.SHOW_FORCED)
     }
 
+    fun setToolbar(activity: FragmentActivity?, title: String, hasUpEnabled: Boolean, show: Boolean = true) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]:
```suggestion
    fun setToolbar(activity: FragmentActivity?, title: String, hasUpEnabled: Boolean = true, show: Boolean = true) {
```

And remove true from calling site

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Okay

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 2]
```diff
@@ -77,6 +79,14 @@ object Utils {
         inputManager.hideSoftInputFromWindow(view.windowToken, InputMethodManager.SHOW_FORCED)
     }
 
+    fun setToolbar(activity: FragmentActivity?, title: String, hasUpEnabled: Boolean = true, show: Boolean = true) {
+        val thisActivity = activity as? AppCompatActivity
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Use smart cast instead.

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 3]
```diff
@@ -77,6 +79,14 @@ object Utils {
...
+        thisActivity?.supportActionBar?.title = title
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: `?` is not required for thisActivity after smart cast.

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 4]
```diff
@@ -77,6 +79,14 @@ object Utils {
...
+        thisActivity?.supportActionBar?.setDisplayHomeAsUpEnabled(hasUpEnabled)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: If the value for `show` is false. Then that logic makes no sense.

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt (hunk 2 above)
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt (hunk 3 above)
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt (hunk 4 above)
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 5]
```diff
@@ -163,10 +155,8 @@ class LoginFragment : Fragment() {
 
         rootView.tick.setOnClickListener {
             rootView.sentEmailLayout.visibility = View.GONE
-            if (thisActivity is AppCompatActivity) {
-                thisActivity.supportActionBar?.show()
-                Utils.navAnimVisible(thisActivity.navigationAuth, requireContext())
-            }
+            setToolbar(activity, getString(R.string.login))
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Use Utils.setToolbar, no extra import needed.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: But, it is okay :+1:

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: okay

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 6]
```diff
@@ -77,6 +79,21 @@ object Utils {
         inputManager.hideSoftInputFromWindow(view.windowToken, InputMethodManager.SHOW_FORCED)
     }
 
+    fun setToolbar(
+        activity: FragmentActivity?,
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Change to Activity

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Okay

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt (hunk 5 above)
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: done

**Comment by iamareebjamal** [comment]
Resolve conflicts @angmas1

**Comment by angmas1** [comment]
Working on it

**Comment by angmas1** [comment]
Resolved.

**Pull request merged**